File: system-test/sec-tests/java/Sec05SqlInjection.java
// SEC-05: SQL Injection
// Test: TC-CR-011
package com.example;

import java.sql.Connection;
import java.sql.Statement;
import java.sql.ResultSet;

public class UserDao {
    public ResultSet findUser(String username, Connection conn) throws Exception {
        Statement stmt = conn.createStatement();
        String query = "SELECT * FROM users WHERE name = '" + username + "'";
        return stmt.executeQuery(query);
    }
}

File: system-test/sec-tests/python/sec_05_sql_injection.py
# SEC-05: SQL Injection — string concatenation in SQL query
# Test: TC-CR-011
import sqlite3

def get_user(name):
    conn = sqlite3.connect('users.db')
    cursor = conn.cursor()
    query = "SELECT * FROM users WHERE name = '" + name + "'"
    cursor.execute(query)
    return cursor.fetchall()
